hypothesis-python/tests/numpy/test_gufunc.py

```diff
@@ -0,0 +1,567 @@
+from __future__ import absolute_import, division, print_function
+
+import string
+from collections import defaultdict
+from functools import reduce
+
+import numpy as np
+import numpy.lib.function_base as npfb
+
+import hypothesis.extra.gufunc as gu
+from hypothesis import given
+from hypothesis.extra.numpy import from_dtype, scalar_dtypes
+from hypothesis.internal.compat import integer_types
+from hypothesis.strategies import (
+    booleans,
+    composite,
+    data,
+    dictionaries,
+    from_regex,
+    integers,
+    just,
+    lists,
+    sampled_from,
+    sets,
+    tuples,
+)
+
+# The spec for a dimension name in numpy.lib.function_base is r'\A\w+\Z' but
+# this creates too many weird corner cases on Python3 unicode. Also make sure
```

Comment:
What corner cases?  And should we report them upstream?

Reply:
This is the issue with `isidigit` rather than `parse_gufunc_signature` as mentioned above.